File: Sort Out The Men From Boys/Sort Out The Men From Boys.dart
List<int> menFromBoys(arr) {
  List<int> even = arr.where((v) => v%2==0).toSet().toList();
  List<int> odd = arr.where((v) => v%2==1).toSet().toList();
  even.sort();
  odd.sort((a,b) => -1 * a.compareTo(b));
  return even + odd;
}

File: Sort Out The Men From Boys/Sort Out The Men From Boys.swift
func menFromBoys(_ arr: [Int]) -> [Int] {
    let even = Set(arr.filter { $0 % 2 == 0 }).sorted()
    let odd = Set(arr.filter { $0 % 2 != 0 }).sorted(by: >)
    return even + odd
}
